File: 2022/Day1CalorieCounting/Solution.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.ArrayList;
import java.util.Scanner;

public class Solution {
    public static void main(String[] args) {
        File inputs = new File("inputs2.txt");
        try {
            Scanner reader = new Scanner(inputs);
            int currentTotal = 0;
            ArrayList<Integer> totals = new ArrayList();
            while(reader.hasNextLine()) {
                String line = reader.nextLine();
                if(line.isEmpty()) {
                    totals.add(currentTotal);
                    currentTotal = 0;
                } else {
                    int num = Integer.parseInt(line);
                    currentTotal += num;
                }
            }
            totals.sort((a, b) -> b - a);
            System.out.println("Largest total: " + totals.get(0));
            System.out.println("Sum of top 3 totals: " + (totals.get(0) + totals.get(1) + totals.get(2)));
        } catch (FileNotFoundException e) {
            System.out.println("Inputs file not found!");
            e.printStackTrace();
        }
        
    }
}

File: 2022/Day1CalorieCounting/Solution.js
fetch('inputs2.txt')
  .then(response => response.text())
  .then(text => {
    console.log(text);
    const lines = text.split('\n');
    const totals = [];
    let currentTotal = 0;
    lines.forEach((line, idx) => {
        let lastIdx = lines.length - 1;
        if(parseInt(line)) {
            const num = parseInt(line);
            currentTotal += num;
            if(idx == lastIdx) {
                totals.push(currentTotal);
            }
        } else {
            totals.push(currentTotal);
            currentTotal = 0;
        }
    });
    totals.sort((a,b) => b - a);
    document.getElementById('largest').innerHTML = totals[0];
    document.getElementById('largest3').innerHTML = totals[2] + totals[1] + totals[0];
  });